x/farming/keeper/genesis.go

```diff
@@ -16,68 +16,66 @@ func (k Keeper) InitGenesis(ctx sdk.Context, genState types.GenesisState) {
 	moduleAcc := k.accountKeeper.GetModuleAccount(ctx, types.ModuleName)
 	k.accountKeeper.SetModuleAccount(ctx, moduleAcc)
 
-	// TODO: decision making is needed
-	// feeCollectorAcc, err := sdk.AccAddressFromBech32(types.DefaultFarmingFeeCollector)
-	// if err != nil {
-	// 	panic(err)
-	// }
-
-	// k.accountKeeper.SetModuleAccount(ctx, feeCollectorAcc)
-
-	// TODO: unimplemented
-	//for _, record := range genState.PlanRecords {
-	//	k.SetPlanRecord(ctx, record)
-	//}
-	//for _, staking := range genState.Stakings {
-	//	k.SetStaking(ctx, staking)
-	//}
-	//for _, reward := range genState.Rewards {
-	//	k.SetReward(ctx, reword)
-	//}
+	_, err := sdk.AccAddressFromBech32(genState.Params.FarmingFeeCollector)
+	if err != nil {
+		panic(err)
+	}
 
+	for _, record := range genState.PlanRecords {
+		plan, err := types.UnpackPlan(&record.Plan)
+		if err != nil {
+			panic(err)
+		}
+		k.SetPlan(ctx, plan)
+		k.SetGlobalPlanId(ctx, plan.GetId())
```

Comment:
What if `PlanRecords` is not sorted correctly? Maybe we should validate that `PlanRecords` is sorted.